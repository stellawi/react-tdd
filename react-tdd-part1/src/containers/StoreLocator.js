import React from 'react';
import Header from '../components/Header';
import Button from '../components/Button';
import Map from '../components/Map';
import mapChooser from '../mapChooser';
import axios from 'axios';

class StoreLocator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentMap: 'none.png',
      shops: [],
    };
  }

  async componentDidMount() {
    const response = await axios.get('http://localhost:3000/data/shops.json');
    this.setState({ shops: response.data.shops });
  }

  chooseMap = (e) => {
    const location = e.target.value;
    this.setState({ currentMap: mapChooser(location) });
  }

  renderButton = () => {
    return (
      this.state.shops.map((shop, id) => <Button
        key={id}
        location={shop.location}
        handleClick={this.chooseMap} />))
  };

  render() {
    return (<div>
      <Header />
      {this.renderButton()}
      <Map imageName={this.state.currentMap} location={this.props.location} />
    </div>);
  }
}

export default StoreLocator;