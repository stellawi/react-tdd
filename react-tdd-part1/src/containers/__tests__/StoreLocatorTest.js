import React from 'react';
import { shallow } from 'enzyme';
import StoreLocator from '../StoreLocator';
import axios from 'axios';
import renderer from 'react-test-renderer';

describe('StoreLocator', () => {
  let mountedStore;
  beforeEach(() => {
    mountedStore = shallow(<StoreLocator />);
  });

  it('called axios.get in componentDidMount', async () => {
    await mountedStore.instance().componentDidMount();
    expect(axios.get).toHaveBeenCalled();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<StoreLocator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('called axios.get with correct url', async () => {
    await mountedStore.instance().componentDidMount();
    expect(axios.get).toHaveBeenCalledWith('http://localhost:3000/data/shops.json');
  });

  it('should update the state of the component', async () => {
    await mountedStore.instance().componentDidMount();
    expect(mountedStore.state()).toHaveProperty('shops', [{
      'location': 'test-location',
      'address': 'test-address',
    }])
  })

  it('should render storelocator on the page', () => {
    let mountedStore = shallow(<StoreLocator />);
  });

  it('should render header on the page', () => {
    expect(mountedStore.find('Header')).toHaveLength(1);
  });

  it('should render a map on the page', () => {
    expect(mountedStore.find('Map')).toHaveLength(1);
  });

  describe('chooseMap', () => {
    it('should update this.state.currentMap when user click on the button', () => {
      let mockEvent = { target: { value: 'everland' } };
      mountedStore.instance().chooseMap(mockEvent);
      expect(mountedStore.instance().state.currentMap).toEqual('everland.png');
    });
  })
})
