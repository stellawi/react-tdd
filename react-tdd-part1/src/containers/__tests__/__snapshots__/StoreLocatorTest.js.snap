// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator should render correctly 1`] = `
<div>
  <header>
    <img
      alt="wired-brain"
      src="images/wired-brain-coffee-logo.png"
    />
  </header>
  <div
    className="MapBox"
  >
    <img
      alt={undefined}
      src="images/none.png"
    />
  </div>
</div>
`;
